### PR TITLE
update / add READMEs

### PR DESCRIPTION
**Pull request by leslie-corbalt**
https://jiraent.cms.gov/browse/CMCSMACD-492

## Purpose

This PR updates the top-level README, adds a top-level service README, and updates all service-specific READMEs.
TODO in future PR:
top-level service README: add how the QuickStart uses Amplify
ui-auth service README: add Amplify's role in OAuth2 authentication flow

Each README is a separate commit.

## Learning

The ui-auth service required a deep-dive into how AWS Cognito works with respect to user pool (as an IdP and backed by Okta), identity pool, user pool client.
The OAuth2 `authentication_code` grant flow has an extension called [PKCE](https://datatracker.ietf.org/doc/html/rfc7636), which adds security to the OAuth2 flow when the client code runs in a browser (e.g., Single-Page App). Apparently, PKCE extension is not supported by Amplify.

TODO: add Amplify info to READMEs

#### Pull Request Creator Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [ ] Someone has been assigned this PR.
- [ ] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/ui-auth/README.md [hunk 1]
```diff
@@ -1,28 +1,115 @@
-# ui-auth
+# The ui-auth Service
 
-## Configuration - AWS Systems Manager Parameter Store (SSM)
+This service:
```
[the remark this answers is not in the document]
Comment (leslie-corbalt) [reply services/ui-auth/README.md:3]: I pulled out the sections below the top-level section and put it here: https://confluenceent.cms.gov/x/uKufEw

**Comment by leslie-corbalt** [comment]
> Oh yeah, one other tip for README PRs: checking the rendered Markdown (you can use VSCode's [Markdown features](https://code.visualstudio.com/docs/languages/markdown)) can help you catch Markdown formatting errors

Just saw this. Thank you! I just installed markdownlint and pushed a few more commits, fixing the lint errors. Now the files should be pretty and lint-free.

File: services/proxy-api/README.md [hunk 2]
```diff
@@ -1,32 +1,18 @@
...
-| .../warmup/concurrency | N | Y | Y | The number of lambda functions to invoke on warmup. The higher this number the warm lambda containers are ready to go.|
+1. /proxyFunc:
+   - POST: forwards the incoming request to the app-api endpoint configured in the SSM parameter `.../prince/api`.
 
-This project uses [AWS Systems Manager Parameter Store](https://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-parameter-store.html), often referred to as simply SSM, to inject environment specific, project specific, and/or sensitive information into the deployment.
-In short, SSM is an AWS service that allows users to store (optionally) encrypted strings in a directory like hierarchy. For example, "/my/first/ssm/param" is a valid path for a parameter. Access to this service and even individual paramters is granted via AWS IAM.
+This service provides a proxy for an app-api endpoint. The proxy service such that it receives an incoming request, extracts the request body, if any, and creates a POST request, with the body, to the proxied endpoint. The proxied endpoint is configurable and stored in SSM parameter store.
```
Comment (ben-harvey) [inline services/proxy-api/README.md:8]: Slight rewording is needed here: 'The proxy service such that it receives...'

File: services/ui-auth/README.md [hunk 3]
```diff
@@ -1,28 +1,36 @@
 # ui-auth
 
-## Configuration - AWS Systems Manager Parameter Store (SSM)
-
-The following values are used to configure the deployment of this service (see below for more background and context).
+This service uses AWS Cognito to authenticate users that sign-in to the QuickStart. Details about how the QuickStart configures the AWS Cognito service are [here](https://confluenceent.cms.gov/x/uKufEw). Changes to how the QuickStart configures Cognito must be copied to [the confluence doc](https://confluenceent.cms.gov/x/uKufEw). _The details are important for QuickStart teams to review. Teams should verify the configuration is correct and secure and appropriate for their application_.
```
Comment (ben-harvey) [inline services/ui-auth/README.md:3]: Revise to 'correct, secure, and appropriate' (if you're an Oxford comma believer like me 😉)

File: services/README.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# services
+
+## Terminology
+
+The term `service` refers to the infrastructure resources and application code defined in a serverless.yml file. The QuickStart app is composed of multiple serverless.yml files; each one is in its own folder, representing a named serice. Information on the syntax of a serverless.yml file, for use with AWS, is [here](https://www.serverless.com/framework/docs/providers/aws/guide/serverless.yml).
```
Comment (ben-harvey) [inline services/README.md:5]: nit: I feel like 'serverless.yml' here and elsewhere should have backticks around it.  But I don't know of any hard and fast rules for backticking, so if you feel otherwise that's fine. I tend to use them for paths, filenames, variable/parameter names, and technical terms.

**Review by ben-harvey — COMMENTED** [review]
Great revisions, Leslie!  I feel like everything is easy to read and has a consistent structure and feel. Just a few more minor copy edits.

File: services/proxy-api/README.md (hunk 2 above)
Comment (leslie-corbalt) [reply services/proxy-api/README.md:8]: Corrected this to: `The proxy service receives...`

File: services/ui-auth/README.md (hunk 3 above)
Comment (leslie-corbalt) [reply services/ui-auth/README.md:3]: Yup. Done.

File: services/README.md (hunk 4 above)
Comment (leslie-corbalt) [reply services/README.md]: Done. Surrounded references to `serverless.yml` with backticks.

**Review by ben-harvey — APPROVED** [review]
LGTM!

**Comment by leslie-corbalt** [comment]
#26
